script.js

```diff
@@ -0,0 +1,250 @@
+
+    newField.innerHTML = `
+        <input type="text" class="form-input kern-functionaliteit flex-grow" placeholder="Functionaliteit ${fieldCount}" required>
+        <button type="button" class="btn-remove btn-danger text-sm px-2 py-1">-</button>
+    `;
+
+    // Add event listener for removing the field
+    const removeButton = newField.querySelector(".btn-remove");
+    removeButton.addEventListener("click", () => {
+        newField.remove();
+        updateFunctionaliteitPlaceholders();
+        updatePreview();
+    });
+
+    kernFunctionaliteitenContainer.appendChild(newField);
+});
+
+// Update Preview
+function updatePreview() {
+    const projectType = document.querySelector("input[name='projectType']:checked")?.value || "Geen projecttype opgegeven";
+    const projectName = document.getElementById("projectName").value || "Geen projectnaam opgegeven";
+    const toelichting = document.getElementById("toelichting").value || "Geen toelichting opgegeven";
+
+    // Retrieve all Kernfunctionaliteiten
+    const kernFunctionaliteitenFields = Array.from(kernFunctionaliteitenContainer.querySelectorAll(".kern-functionaliteit"));
+    const kernFunctionaliteiten = kernFunctionaliteitenFields
+        .map((field, index) => `${index + 1}. ${field.value || "Geen functionaliteit opgegeven"}`)
+        .join("<br>");
+
+    let additionalFields = "";
+
+    // Add Backend or Fullstack specific fields
+    if (projectType === "Backend" || projectType === "Fullstack") {
+        const entities = document.getElementById("entities").value || "Geen entiteiten opgegeven";
+        const formattedEntities = entities.split("\n").map((item, index) => `${index + 1}. ${item}`).join("<br>");
+        additionalFields += `<p><strong>Entiteiten:</strong><br>${formattedEntities}</p>`;
+    }
+
+    // Add Frontend specific fields
+    if (projectType === "Frontend") {
+        const backendType = document.getElementById("backendType").value || "Geen backend geselecteerd";
+        const otherAPIs = document.getElementById("otherAPIs").value || "Geen API opgegeven";
+        additionalFields += `<p><strong>Gekozen backend:</strong> ${backendType}</p>`;
+        additionalFields += `<p><strong>Andere API's:</strong> ${otherAPIs}</p>`;
+    }
+
+    // Update the preview dynamically
+    previewDiv.innerHTML = `
+        <h3 class="text-xl font-bold">${projectName}</h3>
+        <p><strong>Toelichting:</strong><br>${toelichting}</p>
+        <p><strong>Kernfunctionaliteiten:</strong><br>${kernFunctionaliteiten}</p>
+        ${additionalFields}
+    `;
+
+    handleFieldVisibility();
+}
+const validateForm = () => {
+    let isValid = true;
+
+    // Clear previous error messages
+    const errorMessages = document.querySelectorAll('.error-message');
+    errorMessages.forEach((error) => error.remove());
+
+    const displayError = (element, message) => {
```

Comment:
Op alle andere plekken doe je het met innerHTML, dus ik zou OF overal de 'nette' manier doen zoals je hier doet, OF overal `innerHTML` gebruiken